src/main/java/io/cdap/plugin/salesforce/authenticator/Authenticator.java

```diff
@@ -89,12 +102,33 @@ public static OAuthInfo getOAuthInfo(AuthenticatorCredentials credentials) throw
       if (!Strings.isNullOrEmpty(authResponse.getError())) {
         throw new IllegalArgumentException(
           String.format("Cannot authenticate to Salesforce with given credentials. ServerResponse='%s'",
-                  response));
+                        response));
       }
 
       return new OAuthInfo(authResponse.getAccessToken(), authResponse.getInstanceUrl());
     } finally {
       httpClient.stop();
     }
   }
+
+  /**
+   * Set a proxy to http client based on user input.
+   *
+   * @param credentials Credentials contain the proxy server details set in config.
+   * @param httpClient  httpClient to be used to call salesforce APIs
+   */
+  public static void setProxy(AuthenticatorCredentials credentials, HttpClient httpClient) {
+    try {
+      if (!credentials.getProxyUrl().matches(SalesforceConstants.REGEX_PROXY_URL)) {
```

Comment:
nit: Move this out of the try block, since this is a validation.

Reply:
done